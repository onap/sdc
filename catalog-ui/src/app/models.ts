/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by ob0695 on 2/23/2017.
 */
import from = require("core-js/fn/array/from");
export * from './models/activity';
export * from './models/additional-information';
export * from './models/app-config';
export * from './models/validation-config';
export * from './models/plugins-config';
export * from './models/artifacts';
export * from './models/aschema-property';
export * from './models/schema-attribute';
export * from './models/attributes';
export * from './models/capability';
export * from './models/category';
export * from './models/comments';
export * from './models/components/component';
export * from './models/components/displayComponent';
export * from './models/components/resource';
export * from './models/components/service';
export * from './models/componentsInstances/componentInstance';
export * from './models/componentsInstances/resourceInstance';
export * from './models/componentsInstances/serviceInstance';
export * from './models/componentsInstances/serviceProxyInstance';
export * from './models/componentsInstances/serviceSubstitutionInstance';
export * from './models/graph/zones/group-instance';
export * from './models/graph/zones/policy-instance';
export * from './models/graph/zones/zone';
export * from './models/graph/zones/zone-instance';
export * from './models/csar-component';
//export * from './models/data-type-properties';
export * from './models/properties-inputs/property-be-model';
export * from './models/properties-inputs/property-fe-model';
export * from './models/properties-inputs/property-fe-map';
export * from './models/properties-inputs/derived-fe-property';
export * from './models/properties-inputs/property-declare-api-model';
export * from './models/properties-inputs/property-input-detail';
export * from './models/properties-inputs/input-fe-model';
export * from './models/properties-inputs/simple-flat-property';
export * from './models/data-types-map';
export * from './models/data-types';
export * from './models/distribution';
export * from './models/export-excel';
export * from './models/file-download';
export * from './models/graph/graph-links/common-base-link';
export * from './models/graph/graph-links/common-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-simple-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-host-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-ucpe-link';
export * from './models/graph/graph-links/links-factory';
export * from './models/graph/graph-links/module-graph-links/module-ci-link-base';
export * from './models/graph/graph-links/module-graph-links/module-ci-vl-link';
export * from './models/graph/graphTooltip';
export * from './models/graph/assetPopoverObj';
export * from './models/graph/link-menu';
export * from './models/graph/match-relation';
export * from './models/graph/nodes/base-common-node';
export * from './models/graph/nodes/common-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-configuration';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-proxy';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-substitution';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vf';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vfc';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vl';
export * from './models/graph/nodes/modules-graph-nodes/module-node-base';
export * from './models/graph/nodes/nodes-factory';
export * from './models/graph/point';
export * from './models/graph/connectRelationModel';
export * from './models/graph/relationship';
export * from './models/heat-parameters';
export * from './models/input-property-base';
export * from './models/inputs-and-properties';
export * from './models/inputs';
export * from './models/instance-inputs-properties-map';
export * from './models/left-panel';
export * from './models/member';
export * from './models/modules/base-module';
export * from './models/operation';
export * from './models/properties';
export * from './models/requirement';
export * from './models/server-error-response';
export * from './models/tab';
export * from './models/tooltip-data';
export * from './models/user';
export * from './models/validate';
export * from './models/component-metadata';
export * from './models/components-array';
export * from './models/modal';
export * from './models/button';
export * from './models/wizard-step';
export * from './models/radio-button';
export * from './models/filter-properties-assignment-data';
export * from './models/properties-inputs/input-be-model';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/service-instance-properties-and-interfaces';
export * from './models/relationship-types';
export * from './models/tosca-presentation';
export * from './models/node-types';
export * from './models/capability-types';