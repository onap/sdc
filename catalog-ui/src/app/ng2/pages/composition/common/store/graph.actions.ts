export enum SelectedComponentType {
    COMPONENT_INSTANCE = "COMPONENT_INSTANCE",
    GROUP = "GROUP",
    POLICY = "POLICY",
    TOPOLOGY_TEMPLATE = "TOPOLOGY_TEMPLATE"
}

export class UpdateSelectedComponentAction {
    static readonly type = '[COMPOSITION] UpdateSelectedComponent';

        constructor(public payload: {uniqueId?: string, type?: string}) {
    }
}
    
export class SetSelectedComponentAction {
    static readonly type = '[COMPOSITION] SetSelectedComponent';

    constructor(public payload: {component?: any, type?: SelectedComponentType}) {
        }
}

export class OnSidebarOpenOrCloseAction {
    static readonly type = '[COMPOSITION] OnSidebarOpenOrCloseAction';

    constructor() {
    }
}

export class TogglePanelLoadingAction {
    static readonly type = '[COMPOSITION] TogglePanelLoading';
    constructor(public payload: { isLoading: boolean}) {
    }
}
