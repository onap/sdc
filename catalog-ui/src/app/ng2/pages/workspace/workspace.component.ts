/**
 * Created by ob0695 on 6/11/2018.
 */
