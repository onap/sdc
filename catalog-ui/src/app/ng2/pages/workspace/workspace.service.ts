/**
 * Created by ob0695 on 6/5/2018.
 */
/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by rc2122 on 5/23/2017.
 */
import { Injectable } from '@angular/core';
import {WorkspaceMode, ComponentState, Role} from "../../../utils/constants";
import {Component as TopologyTemplate, ComponentMetadata} from "app/models";
import {CacheService} from "../../services/cache.service";
import {IComponentMetadata} from "../../../models/component-metadata";
import {ComponentType} from "../../../utils";

@Injectable()
export class WorkspaceService {

    public metadata:ComponentMetadata;

    constructor(private cacheService:CacheService) {

    }

    public setComponentMetadata = (metadata: ComponentMetadata) => {
        this.metadata = metadata;
    }

    public getMetadataType(): string {
        switch (this.metadata.componentType) {
            case ComponentType.SERVICE:
                return ComponentType.SERVICE;
            default:
                return this.metadata.resourceType;
        }
    }

    public getComponentMode = (component:TopologyTemplate):WorkspaceMode => {//return if is edit or view for resource or service
        let mode = WorkspaceMode.VIEW;

        let user = this.cacheService.get("user");
        if (component.lifecycleState === ComponentState.NOT_CERTIFIED_CHECKOUT &&
            component.lastUpdaterUserId === user.userId) {
            if ((component.isService() || component.isResource()) && user.role == Role.DESIGNER) {
                mode = WorkspaceMode.EDIT;
            }
        }
        return mode;
    }
}

  