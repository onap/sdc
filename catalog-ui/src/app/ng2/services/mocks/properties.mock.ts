/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

// import { PropertiesResponse } from './../../services/responses/properties.response';
//
// export const PROPERTIES_RESPONSE: PropertiesResponse = {
//    "status":"ok",
//    "errorcode":0,
//    "properties": [
//        {"name": "name1"},
//        {"name": "name2"}
//    ]
// };


export const COMPONENT_INSTANCE_RESPONSE: any = [{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.724ef51f-7595-4336-a8ef-b144e1c937da.contrailv2vlansubinterface5","name":"CP2","normalizedName":"cp2","componentUid":"724ef51f-7595-4336-a8ef-b144e1c937da","creationTime":1489586037851,"modificationTime":1489586037851,"posX":"549","posY":"531","propertyValueCounter":1,"inputValueCounter":1,"originType":"CP","customizationUUID":"0222925a-06c6-48ca-8573-13ba3b650539"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.8ac5a229-60d8-4d7f-8e0f-b630da55c3ae.ldsa_os27","name":"VF22","normalizedName":"vf22","componentUid":"8ac5a229-60d8-4d7f-8e0f-b630da55c3ae","creationTime":1489590640842,"modificationTime":1489590640842,"posX":"644","posY":"359","propertyValueCounter":1,"inputValueCounter":1,"originType":"VF","customizationUUID":"fb34e6a1-f0cc-42e3-853f-178fb122d670"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.00ac92c4-a0c5-4567-aefb-f2b7f3fecba9.vl4","name":"VL1","normalizedName":"vl1","componentUid":"00ac92c4-a0c5-4567-aefb-f2b7f3fecba9","creationTime":1489586006630,"modificationTime":1489586006630,"posX":"486","posY":"530","propertyValueCounter":1,"inputValueCounter":1,"originType":"VL","customizationUUID":"8d156a97-38ca-4637-a6a6-9268dd708431"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.39bf293d-2aa9-4596-b85d-a6fdef18aadc.newvf428","name":"VF11","normalizedName":"vf11","componentUid":"39bf293d-2aa9-4596-b85d-a6fdef18aadc","creationTime":1489590742501,"modificationTime":1489590742501,"posX":"350","posY":"76","propertyValueCounter":1,"inputValueCounter":1,"originType":"VF","customizationUUID":"253f646d-87d0-4b84-bc20-20343b6e28a2"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.f8f30484-5761-4de8-9de8-12f288ee0a54.contrailport7","name":"CP1","normalizedName":"cp1","componentUid":"f8f30484-5761-4de8-9de8-12f288ee0a54","creationTime":1489587207447,"modificationTime":1489587207447,"posX":"418","posY":"531","propertyValueCounter":1,"inputValueCounter":1,"originType":"CP","customizationUUID":"206b4ffa-e520-496a-b57b-0c103eff8c17"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.16022d8c-80cf-46e5-a730-5b4f634ea905.extcp38","name":"ExtCP 38","normalizedName":"extcp38","componentUid":"16022d8c-80cf-46e5-a730-5b4f634ea905","creationTime":1489654844116,"modificationTime":1489654844116,"posX":"473","posY":"419","propertyValueCounter":5,"inputValueCounter":1,"originType":"CP","customizationUUID":"21ebc94a-65b9-463e-b696-07cea08f789a"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.f8f30484-5761-4de8-9de8-12f288ee0a54.contrailport44","name":"ContrailPort 44","normalizedName":"contrailport44","componentUid":"f8f30484-5761-4de8-9de8-12f288ee0a54","creationTime":1489939897435,"modificationTime":1489939897435,"posX":"480","posY":"265","propertyValueCounter":1,"inputValueCounter":1,"originType":"CP","customizationUUID":"607e5819-fa19-4aac-97f6-04f092493a02"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.8ac5a229-60d8-4d7f-8e0f-b630da55c3ae.ldsa_os6","name":"VF2","normalizedName":"vf2","componentUid":"8ac5a229-60d8-4d7f-8e0f-b630da55c3ae","creationTime":1489586613957,"modificationTime":1489586613957,"posX":"638","posY":"531","propertyValueCounter":1,"inputValueCounter":1,"originType":"VF","customizationUUID":"12501356-44cc-427b-8749-c114d3434271"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.8ac5a229-60d8-4d7f-8e0f-b630da55c3ae.ldsa_os8","name":"VF1","normalizedName":"vf1","componentUid":"8ac5a229-60d8-4d7f-8e0f-b630da55c3ae","creationTime":1489587393612,"modificationTime":1489587393612,"posX":"334","posY":"529","propertyValueCounter":1,"inputValueCounter":1,"originType":"VF","customizationUUID":"cbca1972-e64c-4119-a430-ec90aa1397a7"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.8ac5a229-60d8-4d7f-8e0f-b630da55c3ae.ldsa_os32","name":"VF12","normalizedName":"vf12","componentUid":"8ac5a229-60d8-4d7f-8e0f-b630da55c3ae","creationTime":1489591429630,"modificationTime":1489591429630,"posX":"367","posY":"269","propertyValueCounter":1,"inputValueCounter":1,"originType":"VF","customizationUUID":"ac9c77c1-e84d-4967-9ea2-d929b56d10a8"}];
export const COMPONENT_INPUT_RESPONSE: any = [{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.extcp38_mac_requirements_mac_range_plan","type":"string","required":true,"definition":false,"description":"reference to a MAC address range plan","password":false,"name":"extcp38_mac_requirements_mac_range_plan","parentUniqueId":"23b78e64-1734-4898-889d-eab9eba50019"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.extcp38_order","type":"integer","required":true,"definition":false,"description":"The order of the CP on the compute instance (e.g. eth2).","schema":{"property":{"definition":true,"password":false}},"password":false,"name":"extcp38_order","parentUniqueId":"23b78e64-1734-4898-889d-eab9eba50019"},{"uniqueId":"23b78e64-1734-4898-889d-eab9eba50019.extcp38_network_role_tag","type":"string","required":true,"definition":false,"description":"Must correlate to the set of defined â€œnetwork-roleâ€? tag identifiers from the associated HEAT template","schema":{"property":{"definition":true,"password":false}},"password":false,"name":"extcp38_network_role_tag","parentUniqueId":"23b78e64-1734-4898-889d-eab9eba50019"}];
export const COMPONENT_PROPERTIES_RESPONSE: any =[{"schema":{"property":{"password":false,"definition":false}},"password":false,"parentUniqueId":"b80765f0-ef2b-43b1-b658-04e13970b5cf","defaultValue":"{\"naming_policy\":\"ggg\",\"ecomp_generated_naming\":false,\"supplemental_data\":{\"fff\":\"44\",\"uuu\":\"56\"}}","name":"inner-map-1","definition":false,"type":"org.openecomp.datatypes.EcompNaming","uniqueId":"property.b80765f0-ef2b-43b1-b658-04e13970b5cf.inner-map-1","required":false},{"schema":{"property":{"password":false,"definition":false}},"password":false,"parentUniqueId":"b80765f0-ef2b-43b1-b658-04e13970b5cf","defaultValue":"{\"min_subnets_count\":1,\"supplemental_data\":{\"aa\":\"11\",\"bb\":\"22\"},\"cidr_mask\":23,\"dhcp_enabled\":false,\"ip_version\":3}","name":"inner-simple-map","definition":false,"type":"org.openecomp.datatypes.network.IPv4SubnetAssignments","uniqueId":"property.b80765f0-ef2b-43b1-b658-04e13970b5cf.inner-simple-map","required":false},{"schema":{"property":{"password":false,"definition":false,"type":"org.openecomp.datatypes.heat.network.AddressPair"}},"password":false,"parentUniqueId":"b80765f0-ef2b-43b1-b658-04e13970b5cf","defaultValue":"{\"aaa\":{\"mac_address\":\"34\",\"ip_address\":\"56\"},\"bbb\":{\"mac_address\":\"sddf\",\"ip_address\":\"dfdg\"}}","name":"data-type-map","definition":false,"type":"map","uniqueId":"property.b80765f0-ef2b-43b1-b658-04e13970b5cf.data-type-map","required":false},{"schema":{"property":{"password":false,"definition":false,"type":"string"}},"password":false,"parentUniqueId":"b80765f0-ef2b-43b1-b658-04e13970b5cf","defaultValue":"{\"www\":\"dfsdf\",\"aaaa\":\"ert\"}","name":"simple-map","definition":false,"type":"map","uniqueId":"property.b80765f0-ef2b-43b1-b658-04e13970b5cf.simple-map","required":false}];

