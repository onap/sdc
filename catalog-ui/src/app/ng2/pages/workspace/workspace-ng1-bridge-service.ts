/**
 * Created by ob0695 on 6/24/2018.
 */
/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */
import {Store} from "@ngxs/store";
import {Injectable} from "@angular/core";
import {UpdateIsViewOnly} from "../../store/actions/workspace.action";

@Injectable()
export class WorkspaceNg1BridgeService {

    constructor(private store: Store) {
    };

    public updateIsViewOnly = (isViewOnly: boolean):void => {
        this.store.dispatch(new UpdateIsViewOnly(isViewOnly));
    }

}
