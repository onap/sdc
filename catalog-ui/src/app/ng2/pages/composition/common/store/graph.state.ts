import { Action, Selector, State, StateContext} from '@ngxs/store';
import {
    OnSidebarOpenOrCloseAction,
    SelectedComponentType,
    SetSelectedComponentAction,
    TogglePanelLoadingAction
} from "./graph.actions";
import { PolicyInstance, GroupInstance, Component as TopologyTemplate, ComponentInstance, LeftPaletteComponent, FullComponentInstance} from "app/models";
import { TopologyTemplateService } from "app/ng2/services/component-services/topology-template.service";
import { tap } from "rxjs/operators";
import { CompositionService } from "app/ng2/pages/composition/composition.service";
import {GroupsService} from "../../../../services/groups.service";
import {PoliciesService} from "../../../../services/policies.service";
import {WorkspaceService} from "../../../workspace/workspace.service";

export class CompositionStateModel {

    isViewOnly?: boolean;
    panelLoading?: boolean;
    selectedComponentType?: SelectedComponentType;
    selectedComponent?: PolicyInstance | GroupInstance | TopologyTemplate | ComponentInstance;
    withSidebar?: boolean;
}

@State<CompositionStateModel>({
    name: 'composition',
    defaults: {
         withSidebar: true
    }
})
export class GraphState {

    constructor(private topologyTemplateService: TopologyTemplateService,
                private compositionService: CompositionService,
         private policiesService:PoliciesService, private groupsService:GroupsService,
                 private workspaceService: WorkspaceService) {}

    @Action(SetSelectedComponentAction)
    setSelectedComponent({dispatch, getState, patchState}:StateContext<CompositionStateModel>, action: SetSelectedComponentAction) {
        
        const state:CompositionStateModel = getState();
        
        patchState({ panelLoading: true });

        if(action.payload.component instanceof ComponentInstance){
            let originComponent = this.compositionService.getOriginComponentById(action.payload.component.getComponentUid());
            if(!originComponent) {
                return this.topologyTemplateService.getFullComponent(action.payload.component.originType, action.payload.component.getComponentUid())
                .pipe(tap(resp => {
                    this.compositionService.addOriginComponent(resp);
                    this.compositionService.setSelectedComponentType(SelectedComponentType.COMPONENT_INSTANCE);
                    patchState({
                        selectedComponent: new FullComponentInstance(action.payload.component, resp), 
                        selectedComponentType: action.payload.type,
                        panelLoading: false
                    });
                    }, err => {
                        patchState({
                            panelLoading: false
                        })
                    }
                ));
            } else {
                patchState({
                    selectedComponent: new FullComponentInstance(action.payload.component, originComponent), 
                    selectedComponentType: action.payload.type,
                    panelLoading: false
                });
            }
        } else if (action.payload.component instanceof PolicyInstance) {
            let topologyTemplate = this.workspaceService.metadata;
            return this.policiesService.getSpecificPolicy(topologyTemplate.componentType, topologyTemplate.uniqueId, action.payload.component.uniqueId).pipe(tap(resp => 
                {
                    this.compositionService.updatePolicy(resp);
                    patchState({
                        selectedComponent: resp, 
                        selectedComponentType: action.payload.type,
                        panelLoading: false
                    })
                }, err => {
                    patchState({
                        panelLoading: false
                    })
                 }
            ));
        
        } else if (action.payload.component instanceof GroupInstance) {
            let topologyTemplate = this.workspaceService.metadata;
            return this.groupsService.getSpecificGroup(topologyTemplate.componentType, topologyTemplate.uniqueId, action.payload.component.uniqueId).pipe(tap(resp => {
                this.compositionService.updateGroup(resp);
                patchState({
                    selectedComponent: resp,
                    selectedComponentType: action.payload.type,
                    panelLoading: false
                });
            }, err => {
                patchState({
                    panelLoading: false
                })
             }
            ));
        } else { //TopologyTemplate
            patchState({
                selectedComponent: action.payload.component,
                selectedComponentType: action.payload.type,
                panelLoading: false
            })
        }
    }


    // @Action(UpdateSelectedComponentNameAction)
    // UpdateSelectedComponentNameAction({patchState}:StateContext<CompositionStateModel>, action: UpdateSelectedComponentNameAction) {

    //     switch(action.payload.type){
    //         case SelectedComponentType.COMPONENT_INSTANCE:
    //             this.store.dispatch(new UpdateComponentInstancesAction([action.payload.component]));
    //             break;
    //         case SelectedComponentType.POLICY:
    //             this.store.dispatch(new UpdatePolicyNameAction(action.payload.uniqueId, action.payload.newName));
    //             break;
    //         case SelectedComponentType.GROUP:
    //             this.store.dispatch(new UpdateGroupInstancesAction)

    //     }
    //     if(action.payload.type === SelectedComponentType.COMPONENT_INSTANCE){
            
    //     } 
    
    // }

    @Selector() 
    static getSelectedComponent(state:CompositionStateModel) {
        return state.selectedComponent;
    }

    @Selector() 
    static getSelectedComponentId(state:CompositionStateModel) {
        return state.selectedComponent.uniqueId;
    }

    @Selector() 
    static getSelectedComponentType(state:CompositionStateModel) {
        return state.selectedComponentType;
    }


    @Action(OnSidebarOpenOrCloseAction)
    onSidebarOpenOrCloseAction({getState, setState}:StateContext<CompositionStateModel>) {
        const state:CompositionStateModel = getState();

        setState({
            ...state,
            withSidebar: !state.withSidebar
        });
    }
    
    @Action(TogglePanelLoadingAction)
    TogglePanelLoading({patchState}:StateContext<CompositionStateModel>, action: TogglePanelLoadingAction) {

        patchState({
            panelLoading: action.payload.isLoading
        });
    }

    @Selector() static withSidebar(state):boolean {
        return state.withSidebar;
    }

}