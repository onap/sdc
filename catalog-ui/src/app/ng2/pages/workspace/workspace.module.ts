/**
 * Created by ob0695 on 6/4/2018.
 */
/**
 * Created by ob0695 on 6/4/2018.
 */
import {NgModule} from "@angular/core";
import {CompositionPageModule} from "../composition/composition-page.module";

import {NgxsModule} from "@ngxs/store";
import {TopologyTemplateService} from "../../services/component-services/topology-template.service";
import {WorkspaceState} from "../../store/states/workspace.state";
import {WorkspaceService} from "./workspace.service";
import {DeploymentPageModule} from "./deployment/deployment-page.module";
import {ToscaArtifactPageModule} from "./tosca-artifacts/tosca-artifact-page.module";
import {InformationArtifactPageModule} from "./information-artifact/information-artifact-page.module";
import { reqAndCapabilitiesModule } from "./req-and-capabilities/req-and-capabilities.module";
import {AttributesModule} from "./attributes/attributes.module";
import {ArtifactsState} from "../../store/states/artifacts.state";
import {InstanceArtifactsState} from "../../store/states/instance-artifacts.state";
import {DeploymentArtifactsPageModule} from "./deployment-artifacts/deployment-artifacts-page.module";
import { DistributionModule } from './disribution/distribution.module';
import { ActivityLogModule } from './activity-log/activity-log.module';

@NgModule({
    declarations: [],
    imports: [
        DeploymentPageModule,
        CompositionPageModule,
        AttributesModule,
        reqAndCapabilitiesModule,
        ToscaArtifactPageModule,
        DeploymentArtifactsPageModule,
        InformationArtifactPageModule,
        DistributionModule,
        ActivityLogModule,
        NgxsModule.forFeature([WorkspaceState, ArtifactsState, InstanceArtifactsState])
    ],

    exports: [],
    entryComponents: [],
    providers: [TopologyTemplateService, WorkspaceService]
})

export class WorkspaceModule {

    constructor() {

    }
}
