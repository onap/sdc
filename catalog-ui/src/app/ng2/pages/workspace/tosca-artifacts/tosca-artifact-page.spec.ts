import {async, ComponentFixture, TestBed} from "@angular/core/testing";
import {NO_ERRORS_SCHEMA} from "@angular/core";
import {ToscaArtifactPageComponent} from "./tosca-artifact-page.component";
import {ConfigureFn, configureTests} from "../../../../../jest/test-config.helper";
import {NgxDatatableModule} from "@swimlane/ngx-datatable";
import {WorkspaceService} from "../workspace.service";
import {SdcUiServices} from "onap-ui-angular";
import {TopologyTemplateService} from "../../../services/component-services/topology-template.service";
import {Observable} from "rxjs/Observable";
import {ComponentMetadata} from "../../../../models/component-metadata";
import 'rxjs/add/observable/of';
import {NgxsModule, Store} from "@ngxs/store";
import {ArtifactsState} from "../../../store/states/artifacts.state";
import {toscaArtifactMock} from "../../../../../jest/mocks/artifacts-mock";

describe('tosca artifacts page', () => {

    let fixture: ComponentFixture<ToscaArtifactPageComponent>;
    let topologyTemplateServiceMock: Partial<TopologyTemplateService>;
    let workspaceServiceMock: Partial<WorkspaceService>;
    let loaderServiceMock: Partial<SdcUiServices.LoaderService>;
    let store: Store;


    beforeEach(
        async(() => {

            topologyTemplateServiceMock = {
                getArtifactsByType: jest.fn().mockImplementation((componentType, id, artifactType) => Observable.of(toscaArtifactMock))
            };
            workspaceServiceMock = {metadata: <ComponentMetadata>{uniqueId: 'service_unique_id', componentType: 'SERVICE'}}

            loaderServiceMock = {
                activate : jest.fn(),
                deactivate: jest.fn()
            }
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [ToscaArtifactPageComponent],
                    imports: [NgxDatatableModule, NgxsModule.forRoot([ArtifactsState])],
                    schemas: [NO_ERRORS_SCHEMA],
                    providers: [
                        {provide: WorkspaceService, useValue: workspaceServiceMock},
                        {provide: TopologyTemplateService, useValue: topologyTemplateServiceMock},
                        {provide: SdcUiServices.LoaderService, useValue: loaderServiceMock }
                    ],
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(ToscaArtifactPageComponent);
                store = testBed.get(Store);
            });
        })
    );

    it('should match current snapshot of tosca artifact pages component', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should see exactly 2 tosca artifacts', () => {
        fixture.componentInstance.ngOnInit();
        fixture.componentInstance.toscaArtifacts$.subscribe((artifacts)=> {
            expect(artifacts.length).toEqual(2);
        })
        store.selectOnce(state => state.artifacts.toscaArtifacts).subscribe(artifacts => {
            expect(artifacts.length).toEqual(9);
        });
    })

});