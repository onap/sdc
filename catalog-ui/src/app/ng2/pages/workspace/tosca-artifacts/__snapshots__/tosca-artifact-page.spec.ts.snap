// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tosca artifacts page should match current snapshot of tosca artifact pages component 1`] = `
<tosca-artifact-page
  serviceLoader={[Function Object]}
  store={[Function Store]}
  table={[Function DatatableComponent]}
  workspaceService={[Function Object]}
>
  <div
    class="tosca-artifact-page"
  >
    <ngx-datatable
      class="ngx-datatable"
      columnmode="flex"
    >
      <div
        visibilityobserver=""
      >
        
        <datatable-body
          class="datatable-body"
        >
          <datatable-selection>
            
            
            
          </datatable-selection>
        </datatable-body>
        
      </div>
    </ngx-datatable>
  </div>
</tosca-artifact-page>
`;
